source/elements/oneDPL/source/parallel_api/iterators.rst

```diff
@@ -301,3 +339,105 @@ operation were applied to each of these iterators. The types ``T`` within the te
 
 ``make_zip_iterator`` constructs and returns an instance of ``zip_iterator``
 using the set of source iterators provided.
+
+.. _iterators-device-accessible:
+
+Customization For User Defined Iterators
+++++++++++++++++++++++++++++++++++++++++
+
+oneDPL provides a mechanism to indicate whether custom iterators are "device accessible content iterators" by defining
+an Argument-Dependent Lookup (ADL) customization point, ``is_onedpl_device_accessible_content_iterator``. oneDPL also
+defines a public trait, ``is_device_accessible_content_iterator[_v]`` to indicate whether an iterator is a "device
+accessible content iterator".
+
+oneDPL queries this information at compile time to determine how to handle iterator types when they are passed to
+algorithms with a ``device_policy`` to avoid unnecessary data movement.
+
+ADL Customization Point: ``is_onedpl_device_accessible_content_iterator``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A free function ``is_onedpl_device_accessible_content_iterator(IteratorT)`` may be defined, which accepts an argument
+of type ``IteratorT`` and returns a type with the characteristics of ``std::true_type`` if ``IteratorT`` refers to
+"device accessible content", or alternatively returns a type with the characteristics of ``std::false_type``
+otherwise. The function must be defined in one of the valid search locations for ADL lookup, which includes the
+namespace of the definition of the iterator type ``IteratorT``.
+
+The function ``is_onedpl_device_accessible_content_iterator`` may be used by oneDPL to determine if the iterator refers
+to "device accessible content" by interrogating its return type at compile-time only. It shall not be called by oneDPL
+outside a ``decltype`` context to determine the return type. Overloads may be provided as forward declarations only,
+without a body defined. ADL lookup is used to determine which function overload to use according to the rules in the
+`C++ Standard`_. Therefore, derived iterator types without an overload for their exact type will match their most
+specific base iterator type if such an overload exists.
+
+The default implementation of ``is_onedpl_device_accessible_content_iterator`` marks the following iterators as to
+"device accessible content iterators":
+* Pointers (to handle USM pointers)
+* Iterators to USM shared allocated ``std::vector``-s when the allocator type is knowable from the iterator type
+* ``std::reverse_iterator<IteratorT>`` when ``IteratorT`` refers to "device accessible content"
+
+
+
+Public Trait: ``oneapi::dpl::is_device_accessible_content_iterator[_v]``
```

Comment:
naming suggestions to consider

- `is_passed_directly_to_device` (perhaps obsolete)
- `is_device_accessible_iterator`
- `is_indirectly_device_accessible`

For the last one it might be `is_device_indirectly_accessible`; not sure what is more accurate from English perspective. But the last proposed option (by me) is my favorite, so far. It also corelates with several C++20 concepts, e.g., `indirectly_writable`

Reply:
I don't really like `is_device_accessible_iterator` because I think there is a difference between the iterator being accessible and its content it refers to being accessible.  However, `is_indirectly_device_accessible` is more clear and concise than what is currently in the PR.

Reply:
I am fine with `is_indirectly_device_accessible`, but we need to get feedback from the customers.

Reply:
I adjusted the name here in the spec.